backend/otel/extract.go

```diff
@@ -0,0 +1,201 @@
+package otel
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	model "github.com/highlight-run/highlight/backend/model"
+	modelInputs "github.com/highlight-run/highlight/backend/private-graph/graph/model"
+	"github.com/highlight-run/highlight/backend/util"
+	"github.com/highlight/highlight/sdk/highlight-go"
+	e "github.com/pkg/errors"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+var fluentProjectPattern = regexp.MustCompile(fmt.Sprintf(`%s=([\S]+)`, highlight.ProjectIDAttribute))
+
+// Extracted fields
+type extractedFields struct {
+	projectID    string
+	projectIDInt int
+	sessionID    string
+	requestID    string
+	source       modelInputs.LogSource
+	serviceName  string
+
+	// The original map data. Should only be used for debugging.
+	// TODO(et) - if we only use this for debugging purposes, should we just omit?
+	resourceAttributes map[string]any
+	spanAttributes     map[string]any
+	eventAttributes    map[string]any
+	scopeAttributes    map[string]any
+	logAttributes      map[string]any
```

Comment:
could we just make these `*map[string]any` so that the struct keeps pointers here rather than copying the whole map? then i'd say it's safe to keep these. otherwise if they are not useful for debugging, we should drop